src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+/* you may not use this file except in compliance with the License.       */
+/* You may obtain a copy of the License at                                */
+/*                                                                        */
+/*     http://www.apache.org/licenses/LICENSE-2.0                         */
+/*                                                                        */
+/* Unless required by applicable law or agreed to in writing, software    */
+/* distributed under the License is distributed on an "AS IS" BASIS,      */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or        */
+/* implied. See the License for the specific language governing           */
+/* permissions and limitations under the License.                         */
+/*                                                                        */
+/* IBM_PROLOG_END_TAG                                                     */
+
+#include <string.h>
+#include <stddef.h>
+
+#include <endian.h>
+
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+
+#include "IBM_Crypto.h"
+#include "IBM_Exception.h"
+#include "IBM_HexString.h"
+#include "IBM_Container.h"
+#include "IBM_Utils.h"
+
+namespace
+{
+    const static uint32_t g_COL_WIDTH = 24;
+
+    const static uint32_t g_UNCOMPRESSED_PUBKEY_FORMAT = 0x04;
+
+    const static std::string g_HASH_ALGO_SHA512_NAME   = "SHA-512";
+    const static std::string g_SIGN_ALGO_ECDSA521_NAME = "ECDSA521";
+
+    template<typename T_PAIR>
+    struct GetKey: public std::unary_function<T_PAIR, typename T_PAIR::first_type>
+    {
+        const typename T_PAIR::first_type& operator()(const T_PAIR& item) const
+        {
+            return item.first;
+        }
+    };
+
+    uint16_t getUint16( const uint8_t *data )
+    {
+        uint16_t value = 0;
+
+        value = data[1] | (data[0] << 8);
+
+        return value;
+    }
+
+    uint32_t getUint32( const uint8_t *data )
+    {
+        uint32_t value = 0;
+
+        value = (data[3] | (data[2] << 8) | (data[1] << 16) | (data[0] << 24));
+
+        return value;
+    }
+
+    uint64_t getUint64( const uint8_t *data )
+    {
+        uint64_t value = 0;
+
+        value = (            data[7]        | ((uint16_t)data[6] << 8)  | 
+                  ((uint32_t)data[5] << 16) | ((uint32_t)data[4] << 24) |
+                  ((uint64_t)data[3] << 32) | ((uint64_t)data[2] << 40) |
+                  ((uint64_t)data[1] << 48) | ((uint64_t)data[0] << 56) );
+
+        return value;
+    }
+
+    void printHexBytes( const uint8_t* bytes, size_t size, size_t leadSpace )
+    {
+        std::vector<uint8_t> byteArray;
+
+        byteArray.assign(bytes, bytes+size);
+
+        IBM_HexString hStr( byteArray );
+
+        hStr.setWidth(36);
+        hStr.setLeadSpace(leadSpace);
+
+        std::cout << hStr;
+    }
+
+  
+    const char* GetHashAlgoName( int p_hashAlgoNum )
+    {
+        static std::map<int, std::string> s_hashAlgoNames;
```

Comment:
This can be compile-time initialized with C++11.

static const std::map<int, std::string> s_hashAlgoNames = { { HASH_ALG_SHA512 , g_HASH_ALGO_SHA512_NAME },  { HASH_ALG_UNKNOWN, "Unknown"s } };